src/protobuf-net/Meta/MetaType.cs

```diff
@@ -2222,8 +2244,40 @@ public bool IsGroup
                 }
                 if (HasReservations) AppendReservations();
                 NewLine(builder, indent).Append('}');
+
+                AddExtraLayerSchemaModels(extraLayeredMembers, beforeMessagePosition);
             }
 
+            void AddExtraLayerSchemaModels(IEnumerable<(ValueMember, string)> wrappedMembers, int pos)
+            {
+                if (wrappedMembers is null || !wrappedMembers.Any()) return;
+                foreach (var (member, originalSchemaType) in wrappedMembers)
+                {
+                    var wrappedSchemaTypeName = GetWrappedSchemaTypeName(originalSchemaType);
+
+                    builder
+                        .NewLine(ref pos, indent)
+                        .Insert("message ", ref pos)
+                        .Insert(wrappedSchemaTypeName, ref pos)
+                        .Insert(" {", ref pos);
+
+                    builder.NewLine(ref pos, indent + 1);
+                    WriteWrappedFieldPayload();
+                    builder.Insert(";", ref pos);
+
+                    builder
+                        .NewLine(ref pos, indent)
+                        .Insert("}", ref pos);
+
+                    void WriteWrappedFieldPayload()
+                    {
+                        builder
+                            .Insert("group ", ref pos, condition: () => member.SupportNull || member.NullWrappedValueGroup)

```

Comment:
I *think* (please check) that `SupportNull` already sets `NullWrappedValueGroup`, so you might not need to test both

Reply:
Unfortunately, we have to check both
![image](https://user-images.githubusercontent.com/31598696/211208170-cf28f9f9-a50d-4957-abfc-30c7fd7d2907.png)